mqtt/mqtt-lwt/src/main/java/enmasse/mqtt/MqttLwt.java

```diff
@@ -129,22 +131,7 @@ private void connect(Future<Void> startFuture) {
 
             Future<ProtonConnection> publishConnFuture = Future.future();
 
-            int port = messagingServicePort;
-            ProtonClientOptions internalConnOptions = this.createClientOptions();
-            if (certDir != null) {
-                port = messagingServiceSecureInternalPort;
-                internalConnOptions
-                        .setSsl(true)
-                        .addEnabledSaslMechanism("EXTERNAL")
-                        .setHostnameVerificationAlgorithm("")
-                        .setPemTrustOptions(new PemTrustOptions()
-                                .addCertPath(new File(certDir, "ca.crt").getAbsolutePath()))
-                        .setPemKeyCertOptions(new PemKeyCertOptions()
-                                .addCertPath(new File(certDir, "tls.crt").getAbsolutePath())
-                                .addKeyPath(new File(certDir, "tls.key").getAbsolutePath()));
-            };
-
-            this.client.connect(internalConnOptions, this.messagingServiceHost, port, done -> {
+            this.client.connect(options, this.messagingServiceHost, options.isSsl() ? messagingServiceAmqpsNormalPort : messagingServicePort, done -> {
```

Comment:
My 2 cents: 'messagingServicePort'/messagingServiceInternalPort are slightly confusing (both should be internal). 

Suggestion:
messagingServiceAmqpBrokerPort,
messagingServiceNormalPort,
messagingServiceAmqpBrokerPortTLS and
messagingServiceNormalPortTLS

Reply:
(I'd probably also consider dropping some or all of messagingService in these, and using the saving to change from BrokerPort to RouteContainerPort)

Reply:
The reason they are named this way is so that they match the environment variables defined by the service. 

However, I think a workaround is to add a mapping in the deployment specification, so I will do that.

Reply:
could be that changing the annotation on the setters is sufficient( spring magic)